circuits/src/field/native/native_gadget.rs

```diff
@@ -771,6 +795,7 @@ where
         _layouter: &mut impl Layouter<F>,
         byte: &AssignedByte<F>,
     ) -> Result<AssignedNative<F>, Error> {
+        self.update_bound(&byte.0, BigUint::from(256u32));
```

Comment:
Why don't you use the cache when you are assigning types such as `Byte` and `Bit`?